src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs

```diff
@@ -293,7 +293,7 @@ private void PrepareConfig(string contentRoot, int port)
 
             ConfigureModuleAndBinding(config.Root, contentRoot, port);
 
-            if (!DeploymentParameters.PublishApplicationBeforeDeployment)
+            if (!DeploymentParameters.PublishApplicationBeforeDeployment && !File.Exists(Path.Combine(contentRoot, "web.config")))
```

Comment:
So we always publish any app that contains web.config?

Reply:
What is the reason?

Reply:
No. The reason we need this check is if a non-published app contains a web.config (e.g. the InProcessTestSite), we don't want to re-add the aspnetcore element. Before, all tests with the inprocess app were published; now they are not.

Reply:
I see, extract a var.